File: android/src/main/java/expo/modules/zigreactnative/ZigReactNativeModule.kt
package expo.modules.zigreactnative

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ZigReactNativeModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ZigReactNative")

    // Add function
    Function("zigAdd") { a: Int, b: Int ->
      zigAdd(a, b)
    }

    // Multiply function
    Function("zigMultiply") { a: Int, b: Int ->
      zigMultiply(a, b)
    }

    // Fibonacci function
    Function("zigFibonacci") { n: Int ->
      zigFibonacci(n)
    }
  }

  companion object {
    init {
      System.loadLibrary("zig_backend")
    }
  }

  // Native method declarations
  private external fun zigAdd(a: Int, b: Int): Int
  private external fun zigMultiply(a: Int, b: Int): Int
  private external fun zigFibonacci(n: Int): Int
}

File: ios/ZigReactNativeModule.swift
import ExpoModulesCore

public class ZigReactNativeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ZigReactNative")

    // Add function
    Function("zigAdd") { (a: Int32, b: Int32) -> Int32 in
      return zig_add(a, b)
    }

    // Multiply function
    Function("zigMultiply") { (a: Int32, b: Int32) -> Int32 in
      return zig_multiply(a, b)
    }

    // Fibonacci function
    Function("zigFibonacci") { (n: Int32) -> Int32 in
      return zig_fibonacci(n)
    }
  }
}
